File: PlankCooking/ClientApp/app/models/OrderCart.ts

export class OrderCart {

    orderCartID: number;
    status: number;
    notes: string;
    purchaseDate: Date;
    history: string;
    taxes: number;
    orderTotal: number;
    shippingFirstName: string;
    shippingLastName: string;
    shippingAddress1: string;
    shippingAddress2: string;
    shippingCity: string;
    shippingState: string;
    shippingPostalCode: string;
    shippingPhone: string;
    shippingEmail: string;
    billingAddress1: string;
    billingAddress2: string;
    billingCity: string;
    billingState: string;
    billingPostalCode: string;
    uniqueIdentifier: string;
    websiteID: number;
    dateCreated: Date;
    billingFirstName: string;
    billingLastName: string;
    billingPhone: string;
    billingEmail: string;
    shippingCountry: string;
    billingCountry: string;
    authorizationCode: string;
    transactionID: string;
    shippingValidation: boolean;
    shippingCost: number;
    shippingType: number;
    creditCardType: string;


    constructor() {

        this.orderCartID = 0;
        this.status = 0;
        this.notes = "";
        this.purchaseDate = new Date;
        this.history = "";
        this.taxes = 0;
        this.orderTotal = 0;
        this.shippingAddress1 = "";
        this.shippingAddress2 = "";
        this.shippingCity = "";
        this.shippingCountry = "";
        this.shippingEmail = "";
        this.shippingFirstName = "";
        this.shippingLastName = "";
        this.shippingPhone = "";
        this.shippingPostalCode = "";
        this.shippingState = "";
        this.billingAddress1 = "";
        this.billingAddress2 = "";
        this.billingCity = "";
        this.billingCountry = "";
        this.billingEmail = "";
        this.billingFirstName = "";
        this.billingLastName = "";
        this.billingPhone = "";
        this.billingPostalCode = "";
        this.billingState = "";
        this.uniqueIdentifier = "";
        this.websiteID = 0;
        this.dateCreated = new Date;
        this.authorizationCode = "";
        this.transactionID = "";
        this.shippingValidation = false;
        this.shippingCost = 0;
        this.shippingType = 0;
        this.creditCardType = "";


    }
}

File: PlankCooking/models/OrderCart.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System;


namespace PlankCooking.Models {

 
 
 public partial class OrderCart
    {
        [Key]
        public int OrderCartID{ get; set; }

        [Required]
        public int  Status { get; set; }

        public string Notes { get; set; }

        [Required, DataType(DataType.DateTime)]
        public DateTime PurchaseDate { get; set; }

        [StringLength(100)]
        public string History { get; set; }

        [DataType(DataType.Currency)]
        public decimal Taxes { get; set; }

        [Required]
        public decimal OrderTotal { get; set; }

        [Required]
        public string ShippingFirstName { get; set; }

        [Required]
        public string ShippingLastName { get; set; }

        [Required]
        public string ShippingAddress1 { get; set; }

        public string ShippingAddress2 { get; set; }

        public string ShippingCity { get; set; }

        public string ShippingState { get; set; }

        [DataType(DataType.PostalCode)]

        public string ShippingPostalCode { get; set; }

        public string ShippingPhone { get; set; }

        public string ShippingEmail { get; set; }

        public string BillingAddress1 { get; set; }

        public string BillingAddress2 { get; set; }

        public string BillingCity { get; set; }

        public string BillingState { get; set; }

         [DataType(DataType.PostalCode)]

        public string BillingPostalCode { get; set; }

        public string UniqueIdentifier {get; set;}

         [ForeignKey("WebsiteID")]
        public Website Website { get; set; }
        public int WebsiteID { get; set; }

        [DataType(DataType.DateTime)]

        public DateTime DateCreated {get; set;}

        public string BillingFirstName {get; set;}

        public string BillingLastName {get; set;}

        public string BillingPhone {get; set;}

        public string BillingEmail {get; set;}

        public string ShippingCountry {get; set;}

        public string BillingCountry {get; set;}

        public string AuthorizationCode {get; set;}

        public string TransactionID {get; set;}

        public bool ShippingValidation {get; set;}

        [DataType(DataType.Currency)]
        public int ShippingCost {get; set;}

        public int ShippingType {get; set;}

        public string CreditCardType {get; set;}

    }
}